src/lib/caching/structures/presences/PresenceBuilder.ts

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable no-dupe-class-members */
+import { PresenceGameBuilder } from './PresenceGameBuilder';
+
+import type { WSIdentify } from '@klasa/ws';
+import type { APIActivityData } from '@klasa/dapi-types';
+
+export type StatusUpdateData = NonNullable<WSIdentify['presence']>;
+export type PresenceStatus = 'online' | 'dnd' | 'idle' | 'invisible' | 'offline';
+
+/**
+ * The presence builder.
+ * @since 0.0.1
+ * @see https://discord.com/developers/docs/topics/gateway#update-status-gateway-status-update-structure
+ */
+export class PresenceBuilder implements StatusUpdateData {
+
+	/**
+	 * Unix time (in milliseconds) of when the client went idle, or null if the client is not idle.
+	 * @since 0.0.1
+	 */
+	public since: number | null;
+
+	/**
+	 * Null, or the user's new activity.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/topics/gateway#activity-object
+	 */
+	public game: APIActivityData | null;
+
+	/**
+	 * The user's new status.
+	 * @since 0.0.1
+	 * @see https://discord.com/developers/docs/topics/gateway#update-status-status-types
+	 */
+	public status: PresenceStatus;
+
+	/**
+	 * Whether or not the client is afk.
+	 * @since 0.0.1
+	 */
+	public afk: boolean;
+
+	public constructor(data: Partial<StatusUpdateData> = {}) {
+		this.since = data.since ?? null;
+		this.game = data.game ?? null;
+		this.status = data.status ?? 'online';
+		this.afk = data.afk ?? false;
+	}
+
+	/**
+	 * Modifies the presence and returns it.
+	 * @since 0.0.1
+	 * @param since Unix time (in milliseconds) of when the client went idle, or null if the client is not idle.
+	 */
+	public setSince(since: number | null): this {
```

Comment:
We should think about defaulting this to Date.now() for function parity with Embed#setTimestamp().